com.ibm.streamsx.topology/com.ibm.streamsx.topology.python/namespace-info.spl

```diff
@@ -0,0 +1,160 @@
+the first positional parameter in the Python function. For example:
+    # SPL input schema: tuple<int32 id, float64 reading>
+    ＠spl.sink
+    def myfunc(a,b):
+      # a is set to: id
+      # b is set to: reading
+
+If more SPL attributes exist than specified parameters then the additional attributes are ignored.
+
+A Python variable argument can be used to capture all of the remaining attributes, for example:
+    # SPL input schema: tuple<int32 id, float64 reading, float lat, float lon>
+    ＠spl.sink
+    def myvarargfunc(a, *values)
+      # a is set to: id
+      # values is set to the values of: reading, lat, lon
+
+# Python function return conversion to SPL Tuples
+A Python function must return `None`, a Python tuple or a list of Python tuples.
+When `None` is return then no tuple will be submitted to the operator's output port. When a single tuple is returned is is converted to an SPL tuple and submitted to the output port. When a list of tuples is returned, each tuple is converted to a tuple and submitted to the output port, in order of the list starting with the first tuple (position 0).
+
+The values of a Python tuple are assigned to an output SPL tuple by position, so the first value in the Python tuple is assigned to the first attribute in the SPL tuple.
+
+    # SPL output schema: tuple<int32 x, float64 y, float32 z>
+    ＠spl.pipe
+    def myfunc(a,b):
+       return a,b,a+b
+
+    # The SPL output will be:
+    # x is set to: a
+    # y is set to: b 
+    # z is set to: a+b
+
+When a returned tuple has less values than attributes in the SPL output schema the attributes not set by the Python function will be set to their SPL default or copied from the input tuple, depending on the operator type.
+
+When a returned tuple has more values than attributes in the SPL output schema then the additional values are ignored.
+
+++ \@spl.pipe
+Decorator to create a pipe SPL operator.
+When a Python function is decorated with `@spl.pipe`
+a pipe SPL operator is created with a single input port
+and single output port.  For each input tuple the function is called,
+and its return value is used to submit zero or more tuples on the output port. 
+
+When the function returns a tuple containing less values than attributes
+in the SPL output schema then the remaining attributes are copied from
+the input tuple if a matching attribute is found, otherwise they are set
+to the SPL default value.
+
+# Examples
+
+Simple `Noop` pipe operator that passes the input SPL tuple onto its output using a variable argument.
+
+    ＠spl.pipe
+    def Noop(*tuple):
+      "Pass the tuple along without any change"
+      return tuple
+
+Simple filter, note that no return statement is equivalent to returning `None`:
+    ＠spl.pipe
+    def SimpleFilter(a,b):
+      "Filter tuples only allowing output if the first attribute is less than the second. Returns the sum of the first two attributes."
+      if (a < b):
+         return a+b,
```

Comment:
nubie python question: is the trailing comma a requirement of python or just our implementation of interpreting the function result?   Requiring it seems inconvenient - i.e., can't we define that a scalar result (`return a+b`) will simply set just the first positional attribute in the SPL typle (just as `return a+b,` will)?